pallets/xcmp-handler/src/migrations/add_flow_to_currency_data.rs

```diff
@@ -0,0 +1,91 @@
+use core::marker::PhantomData;
+
+use codec::{Decode, Encode};
+use frame_support::{
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+	Twox64Concat,
+};
+use scale_info::TypeInfo;
+
+use crate::{Config, XcmCurrencyData, XcmFlow};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+/// Pre-migrations storage struct
+#[derive(Clone, Copy, Debug, Encode, Decode, PartialEq, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct OldXcmCurrencyData {
+	pub native: bool,
+	pub fee_per_second: u128,
+	pub instruction_weight: u64,
+}
+
+impl From<OldXcmCurrencyData> for XcmCurrencyData {
+	fn from(data: OldXcmCurrencyData) -> Self {
+		XcmCurrencyData {
+			native: data.native,
+			fee_per_second: data.fee_per_second,
+			instruction_weight: data.instruction_weight,
+			flow: XcmFlow::Normal,
+		}
+	}
+}
```

Comment:
I see. The flow is now defined in foreign currency data. Ideally, we should detach this chain-level config from currency, but it is a good place and maybe the only place for now.

Reply:
No additional storage read cost within one transaction in `xcm-handler` metadata.